modules/doc_lookup.py

```
@@ -0,0 +1,258 @@
+    try:
+        current = validate_url(url)
+    except LookupRefused as exc:
+        return LookupResult(url=url, ok=False, error=str(exc))
+
+    opener = urllib.request.build_opener(_NoRedirect())
+    for _ in range(MAX_REDIRECTS + 1):
+        request = urllib.request.Request(
+            current,
+            headers={"User-Agent": "repopilot-agent", "Accept": "text/html,text/plain"},
+            method="GET",
+        )
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)